### PR TITLE
docs: restructure bpf guide

### PR DESCRIPTION
**Pull request by yoyo-go**
I restructured the BPF guide, but got below linter errors, I have looked up and didn't find out what's going wrong, the enumerated list has a proper indentation. 
```
Running linter...
./bpfguide/dev_tool.rst:10: (INFO/1) Enumerated list start value not ordinal-1: "3" (ordinal 3)
./bpfguide/reading.rst:10: (INFO/1) Enumerated list start value not ordinal-1: "4" (ordinal 4)
```

### CODE REVIEW

File: Documentation/index.rst [hunk 1]
```diff
@@ -143,5 +143,5 @@ get started and experiment with Cilium.
    :maxdepth: 2
    :caption: BPF and XDP Reference Guide
 
-   bpf
+   bpfguide/index
```
Comment (qmonnet) [inline Documentation/index.rst]: Just `bpf/index` probably, the `guide` part does not need be in the URL

File: Documentation/bpfguide/index.rst [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+   architect
+   dev_tool
+   debug_test
+   program_type
+   reading
+
+
+
+
```
Comment (qmonnet) [inline Documentation/bpfguide/index.rst]: No need to keep all these empty lines at the end of files

File: Documentation/bpfguide/index.rst [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+information. The goal of this chapter is to provide a BPF reference guide in
+order to gain understanding of BPF, its networking specific use including loading
+BPF programs with tc (traffic control) and XDP (eXpress Data Path), and to aid
+with developing Cilium's BPF templates.
+
+.. toctree::
+   :maxdepth: 2
+
+   architect
+   dev_tool
```
Comment (qmonnet) [inline Documentation/bpfguide/index.rst]: s/`dev_tool`/`toolchain`?

File: Documentation/bpfguide/index.rst [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+   debug_test
+   program_type
```
Comment (qmonnet) [inline Documentation/bpfguide/index.rst]: `program_types`, there are several ones, or even `progtypes`?

File: Documentation/bpfguide/index.rst [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+Cilium uses BPF heavily in its data path, see :ref:`ebpf_datapath` for further
+information. The goal of this chapter is to provide a BPF reference guide in
+order to gain understanding of BPF, its networking specific use including loading
+BPF programs with tc (traffic control) and XDP (eXpress Data Path), and to aid
+with developing Cilium's BPF templates.
+
+.. toctree::
+   :maxdepth: 2
+
+   architect
```
Comment (qmonnet) [inline Documentation/bpfguide/index.rst]: An `architect` is a person. `architecture`?

File: Documentation/bpfguide/index.rst [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+   dev_tool
+   debug_test
+   program_type
+   reading
```
Comment (qmonnet) [inline Documentation/bpfguide/index.rst]: `references`? `resources`?

File: Documentation/bpfguide/index.rst [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+order to gain understanding of BPF, its networking specific use including loading
+BPF programs with tc (traffic control) and XDP (eXpress Data Path), and to aid
+with developing Cilium's BPF templates.
+
+.. toctree::
+   :maxdepth: 2
+
+   architect
+   dev_tool
+   debug_test
```
Comment (qmonnet) [inline Documentation/bpfguide/index.rst]: `debug_test` is OK, I'd use `debug_and_test` maybe to make it clear we're not debugging a test.

**Review by qmonnet — CHANGES_REQUESTED** [review]
Looks good, I've got some suggestions on the naming of the files (see below). The directory itself can probably just be `bpf` instead of `bpfguide`.

The CI errors were present before, but we would mask them in the script, this is what we need to adjust here now the files have moved. There's also one line to change in `CODEOWNERS`. Please add the patch below (but this is with `bpfguide/`, you probably want to adjust both lines with `bpf/` if you take my suggestion above):

<details><summary>Patch</summary>

```diff
diff --git a/CODEOWNERS b/CODEOWNERS
index ff22b9f66496..a8f9385ea64e 100644
--- a/CODEOWNERS
+++ b/CODEOWNERS
@@ -267,7 +267,7 @@ Makefile* @cilium/build
 /Documentation/_static/ @cilium/docs-structure
 /Documentation/api.rst @cilium/sig-agent @cilium/docs-structure
 /Documentation/beta.rst @cilium/docs-structure
-/Documentation/bpf.rst @cilium/sig-datapath @cilium/docs-structure
+/Documentation/bpfguide/ @cilium/sig-datapath @cilium/docs-structure
 /Documentation/check-build.sh @cilium/docs-structure
 /Documentation/check-cmdref.sh @cilium/docs-structure
 /Documentation/check-crd-compat-table.sh @cilium/docs-structure
diff --git a/Documentation/check-build.sh b/Documentation/check-build.sh
index bbd2923e286a..220855fb5430 100755
--- a/Documentation/check-build.sh
+++ b/Documentation/check-build.sh
@@ -101,7 +101,7 @@ run_linter() {
     CONF_PY_ROLES=$(sed -n "/^extlinks = {$/,/^}$/ s/^ *'\([^']\+\)':.*/\1/p" conf.py | tr '\n' ',')
     CONF_PY_SUBSTITUTIONS="$(sed -n 's/^\.\. |\([^|]\+\)|.*/\1/p' conf.py | tr '\n' ',')release"
     ignored_messages="("
-    ignored_messages="${ignored_messages}bpf.rst:.*: \(INFO/1\) Enumerated list start value not ordinal"
+    ignored_messages="${ignored_messages}bpfguide/.*\.rst:.*: \(INFO/1\) Enumerated list start value not ordinal"
     ignored_messages="${ignored_messages}|Hyperlink target .*is not referenced\."
     ignored_messages="${ignored_messages}|Duplicate implicit target name:"
     ignored_messages="${ignored_messages}|Malformed table\."
```
</details>

**Review by qmonnet — APPROVED** [review]
Looks good to me, thank you!

**Review by borkmann — APPROVED** [review]

**Pull request merged**